octodns/provider/yaml.py

```diff
@@ -210,23 +317,21 @@ def populate(self, zone, target=False, lenient=False):
             return False
 
         before = len(zone.records)
-        utf8_filename, idna_filename = self.get_filenames(zone)
 
-        # we prefer utf8
-        if isfile(utf8_filename):
-            if utf8_filename != idna_filename and isfile(idna_filename):
-                raise ProviderException(
-                    f'Both UTF-8 "{utf8_filename}" and IDNA "{idna_filename}" exist for {zone.decoded_name}'
-                )
-            filename = utf8_filename
-        else:
-            self.log.warning(
-                'populate: "%s" does not exist, falling back to try idna version "%s"',
-                utf8_filename,
-                idna_filename,
-            )
-            filename = idna_filename
-        self._populate_from_file(filename, zone, lenient)
+        sources = []
+
+        split_extension = self.split_extension
+        if split_extension:
+            sources.extend(self._split_sources(zone))
+
+        if not self.split_extension:
+            sources.append(self._zone_sources(zone))
```

Comment:
Does this mean that a single instance of YamlProvider can't be used to read a setup with a hybrid of non-split and split yaml files? Looks like if I've set `split_extension` to non-empty/non-null, this will skip reading non-split zones altogether.

Reply:
That is sort of correct. 

I was thinking it's kind of confusing to have it look for both a directory named `zone.name./` and a file named `zone.name.yaml` at the same level and it'd be preferable to just look at `zone.name./` and then a single file, with whatever name you want could live under that directory, e.g. `zone.name./zone.name.yaml`, when you don't want to split things up. 

Don't mind adding it back if you'd specifically prefer to be able mix the two and the above isn't to your liking.

Reply:
Yeah I personally prefer support for hybrid setup for the use-case described in https://github.com/octodns/octodns/issues/1040.

Reply:
Reverted that bit. `disable_zonefile` is an option again and defaults to false so when split_extension is enabled it'll also be looking for the standard `zone.name.yaml`